ql/indexes/iborindex.cpp

```diff
@@ -78,9 +78,10 @@ namespace QuantLib {
                                    const Currency& curr,
                                    const Calendar& fixCal,
                                    const DayCounter& dc,
-                                   const Handle<YieldTermStructure>& h)
-   : IborIndex(familyName, 1*Days, settlementDays, curr,
-               fixCal, Following, false, dc, h) {}
+                                   const Handle<YieldTermStructure>& h,
+                                   OvernightAveraging averagingMethod)
```

Comment:
I think the averaging method belongs to the coupon, not to the index.  There's just one index which is fixed every day, whose fixings can be combined differently depending on the instrument.

Reply:
I see your point and fully agree now.

In fact, my initial thinking was to alter the existing `OvernightIndexedCoupon`, by adding a new parameter to the constructor `OvernightAveraging`. Then, based on the value of the enum it would select which coupon pricer to use (similar to how it's currently done in `OvernightIndexFuture`). The tricky bit here is that as a consequence one would have to change the interface of the `OvernightIndexedSwap`, `OISRateHelper`, etc. to propagate this functionality, which feels like a lot...
On the other hand, the instrument construction logic implemented in `averagingois` seems, to my understanding, largely present already in `OvernightIndexedSwap` and `MakeOIS`. So, we only need the flexibility to pick the coupon pricer. Is it the right way to look at it?

Reply:
Hmm. On the one hand, you're right, it's already possible to create the swap, fetch its coupons and call `setCouponPricer` to set whatever pricer you need.  On the other hand, though, it would probably be nice to have the information in the coupon; it feels like it's part of its definition...

Reply:
Ok, how about changing the `OvernightIndexedCoupon` constructor by adding a new parameter:

`        OvernightIndexedCoupon(const Date& paymentDate,
                               Real nominal,
                               const Date& startDate,
                               const Date& endDate,
                               const ext::shared_ptr<OvernightIndex>& overnightIndex,
                               Real gearing = 1.0,
                               Spread spread = 0.0,
                               const Date& refPeriodStart = Date(),
                               const Date& refPeriodEnd = Date(),
                               const DayCounter& dayCounter = DayCounter(),
                               bool telescopicValueDates = false,
                               OvernightAveraging averagingMethod=OvernightAveraging::Compound);`

Would that be something worth considering?

Reply:
Sure.

Reply:
It could bubble up to the instruments using it, as well.

Reply:
Ok great, I'll give it a try.